procedure/pull-request-merge.rst

```diff
@@ -17,133 +17,457 @@ Memory Pool System pull request merge procedure
 This document contains a procedure for merging a branch received via a
 GitHub "pull request".
 
-This is a draft procedure based on:
-- "Re: Possible MPS Help" [GDR_2014-01-09]_
-- "Memory Pool System branching and merging procedures" [GDR_2020-09-03]_.
+Time to execute:
 
-Some of the questions that need resolving are noted in square
-brackets.
+- first time: 2 hours
+- second time: 30 minutes
+- with practice: < 10 minutes
 
-Ravenbrook is currently (2023-01) migrating the MPS project to git
-(and GitHub) and this is likely to modify this procedure.
+these are measurements_ taken during simple merges that passed automated tests.
 
+.. _measurements: https://github.com/Ravenbrook/mps/pull/97#issuecomment-1381771818
 
-2. Pre-merge checklist
+Ravenbrook is currently (2023-01) `migrating the MPS project to git
+(and GitHub) <https://github.com/Ravenbrook/mps/issues/98>`_ and that
+will greatly simplify this procedure.
+
+This procedure assumes a pull request has been received via GitHub,
+but that dependency is light.  The procedure can be varied for other
+sources, even where the "pull request" is received by email or some
+other traceable document.
+
+This document was created as a combination of the process improvement
+steps from our Perforce-based "Memory Pool System branching and
+merging procedures" [GDR_2014-01-09]_ with Gareth Rees' email
+containing the working steps for git / GitHub merges, "Re: Possible
+MPS Help" [GDR_2020-09-03]_ .
+
+
+2. Purpose
+----------
+
+The purpose of this procedure is:
+
+1. get required changes in to the MPS
+
+2. protect the MPS from introduction of defects
+
+3. maintain and protect the permanent record of the MPS in Perforce
+   and Git, so that defects can be discovered, fixed, and prevented
+
+As with any procedure, you can vary this one to meet this purpose, but
+you should probably read section "`6. Rationale`_".
+
+
+3. Pre-merge checklist
 ----------------------
 
-#. Is there a document (issue, job) recording the problem that is
-   solved by the changes in the branch?
+The answers to these questions should be "yes".  If the answer to a
+question isn't "yes", record that, and why, in response to the pull
+request (and maybe suggest what to do about it).  Include a permalink
+to the revision of this procedure that you are following.  When you
+finish the checklist, decide whether to start `the merging procedure`_
+by considering `2. Purpose`_.
+
+#. Is there a permanent visible document (issue, job), referenced by
+   the branch, recording the problem that is solved by the changes in
+   the branch?
 
 #. Does the branch solve the problem?
 
-#. Is there an automated test case that demonstrates that the problem
-   is solved?
+#. Is there an automated test case that demonstrates the problem
+   (without the branch) and that that the problem is solved (with the
+   branch)?
+
+#. If there are interface changes, are they documented?
```

Comment:
@UNAA008: Interfaces to what?

Reply:
Changes to the MPS interface, as defined by the documentation. I'll clarify.

Reply:
Clarified and linked to manual definition in 6fd5a3d2d